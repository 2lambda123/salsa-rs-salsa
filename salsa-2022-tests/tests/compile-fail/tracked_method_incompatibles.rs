#[salsa::jar(db = Db)]
struct Jar(Tracked<'_>);

#[salsa::tracked(jar = Jar)]
struct Tracked<'db> {
    field: u32,
}

impl<'db> Tracked<'db> {
    #[salsa::tracked]
    fn use_tracked(&self) {}
}

trait Db: salsa::DbWithJar<Jar> {}

fn main() {}
