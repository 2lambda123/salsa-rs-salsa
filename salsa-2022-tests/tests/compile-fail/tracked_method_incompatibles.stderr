error: tracked functions must have at least a database and salsa struct argument
  --> tests/compile-fail/tracked_method_incompatibles.rs:13:8
   |
13 |     fn use_tracked(&self) {
   |        ^^^^^^^^^^^
